From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Active ESL Android CI <engineering@active-esl.com>
Date: Fri, 18 Sep 2026 09:45:00 +0100
Subject: [PATCH] vehicle: Allow NXP vendor HAL to use vehicle protobuf

NXP's Android 16 vehicle implementation links the existing vehicle 2.0
protobuf library. Soong permits a module outside //vendor to grant access
to vendor consumers only through //vendor:__subpackages__. Match the NXP
Android 16 hardware-interfaces source's visibility rule.

---
 automotive/vehicle/proto/Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/automotive/vehicle/proto/Android.bp b/automotive/vehicle/proto/Android.bp
--- a/automotive/vehicle/proto/Android.bp
+++ b/automotive/vehicle/proto/Android.bp
@@ -28,6 +28,7 @@ cc_library_static {
         "//hardware/interfaces/automotive/vehicle:__subpackages__",
         "//device/generic/car/emulator:__subpackages__",
         "//system/software_defined_vehicle/core_services:__subpackages__",
+        "//vendor:__subpackages__",
     ],
     vendor: true,
     host_supported: true,
-- 
2.43.0
